src/components/chat-item/chat-prompt-input.ts

```diff
@@ -312,10 +322,47 @@ export class ChatPromptInput {
 
           this.quickPickOpen = true;
         }
+      } else if (navigationalKeys.includes(e.key)) {
+        const direction = e.key === KeyMap.ARROW_UP ? 'up' : 'down';
+        this.clearTextArea();
+
+        if (this.userPromptHistoryIndex === -1) {
+          this.userPromptHistoryIndex = this.userPromptHistory.length;
+        }
+
+        if (direction === 'up') {
+          this.userPromptHistoryIndex = Math.max(0, this.userPromptHistoryIndex - 1);
+        } else if (direction === 'down') {
+          this.userPromptHistoryIndex = Math.min(this.userPromptHistory.length, this.userPromptHistoryIndex + 1);
+        }
+
+        if (this.userPromptHistoryIndex === this.userPromptHistory.length) {
+          this.promptTextInput.updateTextInputValue('');
+        } else {
+          this.promptTextInput.updateTextInputValue(this.userPromptHistory[this.userPromptHistoryIndex].inputText);
+          // use the addAttachment function
+          let codeAttachment = this.userPromptHistory[this.userPromptHistoryIndex].codeAttachment;
+          if (typeof codeAttachment === 'string' && codeAttachment.trim().length > 0) {
+            codeAttachment = codeAttachment
+              .replace(/~~~~~~~~~~/, '')
+              .replace(/~~~~~~~~~~$/, '')
+              .trim();
+            this.promptAttachment.updateAttachment(codeAttachment, 'code');
+          }
+
+          const promptLength = this.promptTextInput.getTextInputValue().trim().length + (typeof codeAttachment === 'string' ? codeAttachment.length : 0);
+          this.promptTextInput.updateTextInputMaxLength(Math.max(MAX_USER_INPUT_THRESHOLD, (MAX_USER_INPUT() - promptLength)));
```

Comment:
To update the text input max length I consider both attachment and current input text

Reply:
So the things you'll get from the input and the attachment were already being calculated before they went into the elements. This recalculation seems not necessary. Also the same for replacements, they're already being applied.

Reply:
without replacement the code attachment will show '~~~~~~~~~~'

Reply:
With recalculation you mean updating via `updateTextInputMaxLength` and `updateAvailableCharactersIndicator` right?